dashboard/config/scripts_json/csp1-2022.script_json

```diff
@@ -122,7 +122,7 @@
       "properties": {
         "creative_commons_license": "Creative Commons BY-NC-SA",
         "overview": "Students will create rules for ordering patterns of circles and squares. Students generate all possible messages with three place values, then create rules that explain how they ordered each message. Emphasis is placed on creating clear rules so that, if another group were to follow the rules, they would generate the same list in the same order. Using these rules, students then try to list all possible messages with four place values. As the lesson concludes, students share their rules with classmates.",
-        "preparation": "* Have scissors ready for groups to create [r ab-cutouts/csp/2021] or have these pre-cut and prepared before class\r\n",
+        "preparation": "* Have scissors ready for groups to create [r shape_cutouts/csp/2022] or have these pre-cut and prepared before class\r\n",
```

Comment:
why is the resource key ('ab-cutouts') changing here -- are we regenerating the key from the resource name when we copy?

Reply:
It looks like [we are](https://github.com/code-dot-org/code-dot-org/blob/1d2eeb44d8673a37dfdee62ba5588bc7a98fb621/dashboard/app/models/resource.rb#L140). I double checked this specific change and the [2022](https://github.com/code-dot-org/code-dot-org/blob/1d2eeb44d8673a37dfdee62ba5588bc7a98fb621/dashboard/config/scripts_json/csp1-2022.script_json#L5939) and [2021](https://github.com/code-dot-org/code-dot-org/blob/1d2eeb44d8673a37dfdee62ba5588bc7a98fb621/dashboard/config/scripts_json/csp1-2021.script_json#L5744) resources do look the same.

We could definitely change the `copy_to_course_version` logic to try to preserve the key if we wanted.

Reply:
thank you for the pointer! the current behavior sounds good to me.